### DISCUSSION TITLE
`writeContract` throws

### DISCUSSION THREAD

**Discussion by gershido** [wevm/viem#732] [Question] [answered]
Hey :) Using a local account (private key), `writeContract` throws an error in case the transaction will revert. I was not expecting this, as I thought an error will be thrown only when using `waitForTransactionReceipt`. It seems that the error is thrown on the gas estimation... This makes sense but I'm asking to make sure that this is intended. Thank you!

**Comment by jxom** [comment, marked as answer]
Yes this is expected. 😊 

You can also bypass the gas estimation check if you pass your own `gas` value.

**Reply by ssuvorov** [reply]
if I pass gas: 800000 it fails with `"ContractFunctionExecutionError: Execution reverted for an unknown reason.".`
but adding one extra 0 > 8000000 allows the transaction to pass.
If I won't provide the gas value, it fails with 
`The total cost (gas * gas fee + value) of executing this transaction exceeds the balance of the account.`.
Maybe you can help here: what's a proper way to solve that?

**Comment by gershido** [comment]
Regarding getting the revert reason in that case (thrown on gas estimation), is it possible? What I get is:
"EstimateGasExecutionError: Execution reverted for an unknown reason", while the revert on the contract does have a message.

**Reply by ssuvorov** [reply]
I spent so much time on that. I have even a better error: `ContractFunctionExecutionError: Execution reverted for an unknown reason."`
Such a weird explanation that doesn't give you any direction on what to check

**Reply by tmm** [reply]
Suggestions welcome!

**Reply by ssuvorov** [reply]
It might be that the contract itself is rejected with this error. Is it possible somehow to know the source of the error? At first I thought it was a bug in wagmi or viem.
